Content.Shared/_RMC14/Atmos/SharedRMCFlammableSystem.cs

```diff
@@ -339,6 +342,171 @@ private void OnFlammableIgnite(Entity<FlammableComponent> ent, ref RMCIgniteEven
+        }
+    }
+
+    private void OnIgniteEndCollide(Entity<RMCIgniteOnCollideComponent> ent, ref EndCollideEvent args)
+    {
+        UntrackFireContact(ent.Owner, args.OtherEntity);
+
+        if (!IsEntityTouchingFire(args.OtherEntity))
+        {
+            RemCompDeferred<SteppingOnFireComponent>(args.OtherEntity);
+            _entitiesNeedingRecheck.Remove(args.OtherEntity);
+        }
+    }
+
+    private void OnIgniteDamageCollide(Entity<RMCIgniteOnCollideComponent> ent, ref DamageCollideEvent args)
+    {
+        Ignite(args.Target, ent.Comp.Intensity, ent.Comp.Duration, ent.Comp.MaxStacks);
+    }
+
+    private void OnIgniteComponentInit(Entity<RMCIgniteOnCollideComponent> ent, ref ComponentInit args)
+    {
+        Timer.Spawn(TimeSpan.FromMilliseconds(50), () => CheckForExistingOverlaps(ent));
+    }
+
+    private void OnIgniteRemoved(Entity<RMCIgniteOnCollideComponent> ent, ref ComponentRemove args)
+    {
+        if (_fireContacts.TryGetValue(ent.Owner, out var contacts))
+        {
+            foreach (var contact in contacts)
+            {
+                if (!IsEntityTouchingFire(contact))
+                {
+                    RemCompDeferred<SteppingOnFireComponent>(contact);
+                    _entitiesNeedingRecheck.Remove(contact);
+                }
+            }
+        }
+        _fireContacts.Remove(ent.Owner);
+    }
+
+    private void OnExtinguishStartCollide(Entity<ExtinguishFireComponent> ent, ref StartCollideEvent args)
+    {
+        if ((args.OurFixture.CollisionLayer & (int)ent.Comp.Collision) == 0)
+            return;
+
+        TrackExtinguishContact(ent.Owner, args.OtherEntity);
+        TryExtinguishEntity(ent.Owner, args.OtherEntity);
+    }
+
+    private void OnExtinguishEndCollide(Entity<ExtinguishFireComponent> ent, ref EndCollideEvent args)
+    {
+        UntrackExtinguishContact(ent.Owner, args.OtherEntity);
+    }
+
+    private void OnExtinguishRemoved(Entity<ExtinguishFireComponent> ent, ref ComponentRemove args)
+    {
+        _extinguishContacts.Remove(ent.Owner);
+    }
+
+    private void CheckForExistingOverlaps(Entity<RMCIgniteOnCollideComponent> ent)
+    {
+        if (TerminatingOrDeleted(ent.Owner))
+            return;
+
+        var intersecting = _physics.GetEntitiesIntersectingBody(ent.Owner, (int)ent.Comp.Collision);
+
+        foreach (var entity in intersecting)
+        {
+            TrackFireContact(ent.Owner, entity);
+            TryIgnite(ent, entity, false);
+            if (_flammableQuery.HasComp(entity))
+            {
+                EnsureComp<SteppingOnFireComponent>(entity);
+                _entitiesNeedingRecheck.Add(entity);
+            }
+        }
+    }
+
+    private void TrackFireContact(EntityUid fireSource, EntityUid target)
+    {
+        if (!_fireContacts.TryGetValue(fireSource, out var contacts))
+        {
+            contacts = new HashSet<EntityUid>();
+            _fireContacts[fireSource] = contacts;
```

Comment:
This set should be stored on a component on the fireSource entity, static data for entities in entity systems is usually a bad idea, specially since it's not getting cleared on round restart cleanup event in this
Same with _extinguishContacts